cpp/src/join/join.cu

```diff
@@ -134,23 +135,24 @@ std::unique_ptr<table> left_join(table_view const& left_input,
   table_view const left  = scatter_columns(matched.second.front(), left_on, left_input);
   table_view const right = scatter_columns(matched.second.back(), right_on, right_input);
 
-  auto join_indices = left_join(left.select(left_on), right.select(right_on), compare_nulls);
+  auto const [left_join_indices, right_join_indices] = cudf::detail::left_join(
+    left.select(left_on), right.select(right_on), compare_nulls, stream, mr);
 
   if ((left_on.empty() || right_on.empty()) ||
```

Comment:
It looks like this `if` statement is for an early exit with an empty result. However, we're calling `cudf::detail::left_join` before the early-exit. None of the early exit conditions `left_on.empty()`, `right_on.empty()`, or `is_trivial_join(left, right, ...)` appear to depend on the result of the call to `cudf::detail::left_join`.

It seems like we could move these lines around, so that the `if` statement and early exit could be evaluated before calling `cudf::detail::left_join`. (Am I missing something important?)

Reply:
Good catch! And tbh no need to apologize at all. I'm working on improving the hash join performance https://github.com/rapidsai/cudf/pull/10248 thus found small issues here and there. That's also why this PR exists :smile:. The goal is to improve hash join!